onnx/utils.py

```diff
@@ -40,25 +39,16 @@ def _build_output_dict(graph) -> dict[str, int]:
                 output_to_index[output_name] = index
         return output_to_index
 
-    def _collect_new_io_core(
-        self,
-        original_io: RepeatedCompositeFieldContainer[ValueInfoProto],
-        io_names_to_extract: list[str],
-    ) -> list[ValueInfoProto]:
-        original_io_map = self._build_name2obj_dict(original_io)
-        new_io_tensors = []
-        for io_name_to_extract in io_names_to_extract:
-            if io_name_to_extract in original_io_map:
-                new_io_tensors.append(original_io_map[io_name_to_extract])
-            else:
-                new_io_tensors.append(self.vimap[io_name_to_extract])
-        return new_io_tensors  # same order as io_names_to_extract
-
-    def _collect_new_inputs(self, names: list[str]) -> list[ValueInfoProto]:
-        return self._collect_new_io_core(self.graph.input, names)
-
-    def _collect_new_outputs(self, names: list[str]) -> list[ValueInfoProto]:
-        return self._collect_new_io_core(self.graph.output, names)
+    def _collect_new_io(self, io_names_to_extract: list[str]) -> list[ValueInfoProto]:
+        # Validate that all names exist in self.value_infos
+        missing_names = [
+            name for name in io_names_to_extract if name not in self.value_infos
+        ]
+        if missing_names:
+            raise ValueError(
+                f"The following names were not found in value_infos: {', '.join(missing_names)}"
```

Comment:
IIRC there are many ONNX models violating this. Not sure if they are generated by some legacy ONNX exporters.

Reply:
In any case this error message makes it clearer to users what's going wrong. Otherwise there will just be a key error